lib/instrumenter/collectors-gz/zpool.js

```diff
@@ -134,6 +151,11 @@ function getZpoolStats(callback) {
                 metric = mod_jsprim.deepCopy(template);
                 metric.key = 'zpool_' + keys[ind].key + '_' + keys[ind].unit;
                 metric.value = z[ind].trim();
+
+                if (keys[ind].encoder) {
+                    metric.value = keys[ind].encoder(metric.value);
```

Comment:
I just have a couple questions here:
- what if the encoder returns -1 like the error case for `zpoolHealthEnc`? Maybe we can assert that all encoders return >= 0, and a negative return value means something undefined happened? Or is it valuable to return -1 in the formatted metrics?
- I'm guessing we don't need to run `trim()` on these since we're relying on another function to give us the string to use. Maybe we can wrap the `z[ind].trim()` line in an `else` block to make that clear.

Reply:
- Yes, `-1` means `zpoolHealthEnc()` failed to encode pool health status. This should happen if ZFS code is updated and a new pool status is introduced. I updated the code to print a warning message if that happens. However, that doesn't mean we should abort the agent or stop processing this metric. IMO, `-1` tells the user that the pool is not online and we don't know its current status. I think returning `-1` is better than returning no metric.

- Assuming `zpool list -Hp` output doesn't contain spaces, I don't think we don't need `z[ind].trim()`, I removed it from the code. I don't know why it was there in the first place? I don't think we need an `else` block. We should be able to use `zpool list` output as it is for metrics with no encoders.